### PR TITLE
[WIP] (Maybe) detuple struct tuples

### PR DESCRIPTION
**Pull request by cartermp**
This is inspired by the following code:

```fsharp
let rec runWithTuple t offset times =
    let offsetValues x y z offset =
        (x + offset, y + offset, z + offset)
    if times <= 0 then
        t
    else
        let (x, y, z) = t
        let r = offsetValues x y z offset
        runWithTuple r offset (times - 1)
let rec runWithStructTuple t offset times =
    let offsetValues x y z offset =
        struct(x + offset, y + offset, z + offset)
    if times <= 0 then
        t
    else
        let struct(x, y, z) = t
        let r = offsetValues x y z offset
        runWithStructTuple r offset (times - 1)
```

`runWithStructTuple ` is faster than `runWithTuple ` for any decently large value of `times` by about 40% in benchmarking. But if you mark the inner functions as `inline`, then `runWithTuple ` is actually marginally faster because some optimizations kick in.

Maybe this fixes it, maybe it doesn't. Let's see what CI says

### CODE REVIEW

**Comment by cartermp** [comment]
lmao nothing failed

File: src/fsharp/DetupleArgs.fs [hunk 1]
```diff
@@ -360,14 +360,19 @@ let checkTS = function
...
         TupleTS (List.map uncheckedExprTS args)
     | _ -> 
         UnknownTS
 
 let rec uncheckedTypeTS g ty =
-    if isRefTupleTy g ty then 
-        let tys = destRefTupleTy g ty 
+    if isAnyTupleTy g ty then 
+        let tys =
+            if isRefTupleTy g ty then
```
Comment (forki) [inline src/fsharp/DetupleArgs.fs]: Those 4 lines are introduced 3 times. Better to put that into a destAnyTuppeTy

Comment (cartermp) [reply src/fsharp/DetupleArgs.fs]: yerp.

`destAnyTuplé`

**Comment by cartermp** [comment]
This one's more complicated.

**Pull request closed without merging**